File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int maxScope = 0;
        int maxLeft = values[0];
        for (int i = 1; i<values.length; i++){
            maxScope = Math.max(maxScope, maxLeft+values[i] - i);
            maxLeft = Math.max(maxLeft, values[i]+i);
        }
        return maxScope;
    }
}

File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.cs
public class Solution {
    public int MaxScoreSightseeingPair(int[] values) {
        int maxLeft = values[0];
        int maxScore = 0;

        for (int j = 1; j < values.Length; j++) {
            maxScore = Math.Max(maxScore, maxLeft + values[j] - j);
            maxLeft = Math.Max(maxLeft, values[j] + j);
        }

        return maxScore;
    }
}